### PR TITLE
5088: Add new status filter for timeline table

### PR DESCRIPTION
**Pull request by X0nic**

### CODE REVIEW

File: app/controllers/admin/loans_controller.rb [hunk 1]
```diff
@@ -165,8 +165,10 @@ def prep_form_vars
   def prep_timeline
     filters = {}
     filters[:type] = params[:type] if params[:type].present?
+    filters[:status] = params[:status] if params[:status].present?
     @loan.root_timeline_entry.filters = filters
     @type_options = ProjectStep.step_type_option_set.translated_list
+    @status_options = [ ['Finalized', 'finalized'], ['Incomplete', 'incomplete'], ['Complete', 'complete'] ]
```
Comment (X0nic) [inline app/controllers/admin/loans_controller.rb]: I need help with creating a `translated_list` like `ProjectStep.step_type_option_set`

Comment (X0nic) [inline app/controllers/admin/loans_controller.rb]: There is also no Draft state yet, so I am using Finalized for now.

File: app/controllers/admin/loans_controller.rb [hunk 2]
```diff
@@ -165,8 +165,12 @@ def prep_form_vars
   def prep_timeline
     filters = {}
     filters[:type] = params[:type] if params[:type].present?
+    filters[:status] = params[:status] if params[:status].present?
     @loan.root_timeline_entry.filters = filters
     @type_options = ProjectStep.step_type_option_set.translated_list
+    @status_options = ProjectStep::COMPLETION_STATUSES.map do |status|
+      [ I18n.t("project_step.completion_status.#{status}"), status ]
+    end.sort
```
Comment (hooverlunch) [inline app/controllers/admin/loans_controller.rb]: Nice. Though you should need to sort these. Just go with a sensible order in the constant.

File: app/views/admin/loans/timeline/_top_controls.html.slim [hunk 3]
```diff
@@ -30,6 +30,11 @@ header.timeline-header.row
         include_blank: I18n.t("project_step.all_types"),
         class: "form-control"
 
+      = select_tag :status,
+        options_for_select(@status_options),
+        include_blank: "All Statuses",
```
Comment (hooverlunch) [inline app/views/admin/loans/timeline/_top_controls.html.slim]: Please i18n the blank option as with the above one.

**Pull request merged**